### ISSUE TITLE
VRC-LOG not submitting consistanly 

### ISSUE THREAD

**Issue by Chayotea** [ShayBox/VRC-LOG#6]
Was shrugging it off as maybe avatars were simply blacklisted but it was happening too often. It is printing out ID's like normal. Copied them into VRCX and way too many were not in VRCDB database. I then uploaded a public avatar to confirm it. It has been a week that I uploaded it and it still is not in the database. Renamed my cache folder so a new one is created and ran vrchat with the public avatar I uploaded and still nothing. Jumped to a public world and one of the other avatars in the lobby appeared in the most recent in the database... most likely submitted by me because I still see that I failed to log my own avatar since I was still wearing it. Downgrading to any other version other than the current one 0.6.0 results in VRC-LOG printing out a single ID before simply closing. The public test avatar I made is just a cube with a texture. 

Is it possible that it is submitting too fast and locally does not try to re-submit until the 30 day cool-down? Though old versions of VRC-LOG now simply crash so that Is also suspicious.

**Comment by ShayBox** [comment]
I believe the API has changed from 201 for unique avatars, to 404 for avatar not found. They will have still been submitted, but not added to cache or shown in logs as submitted.

Try this version https://github.com/ShayBox/VRC-LOG/releases/tag/0.6.1

**Comment by Chayotea** [comment]
OK tested it for a while and I get a successful submit most of the time but often I see this.

vrcx://avatar/avtr_fb77619d-a0e5-4c3c-b906-5930e78dd6d6
[Avatar Search] 500

I checked and it is not in the database. You can see how often it seems to happen. The large amount of generic 500 errors at the end is when I visited Gewehr 41's avatar world.

[VRClog.txt](https://github.com/user-attachments/files/15971470/VRClog.txt)

**Comment by ShayBox** [comment]
Okay, I think there might just be an issue with the API, I think the 500s are successful unique avatars, but the code server to get the avatar info is failing, while 404 means invalid. I'll report back when I find out more.

**Comment by Chayotea** [comment]
Noticing the 7.1 release notes did something change at all on the API side? Yesterday I tried submitting my test avatar using the discord bot and it failed again. Keep in mind the first time it failed was in June. I have never seen it go from "pending" for a day to submitted. Months ago I suspected it was broken. I also noticed the few messages you left on the Avatar Search Discord of which neither seem to result in a reply. My suspicions is that this has been going on since the change of the new API all the way back when Ravenwood was shutdown.

**Comment by ShayBox** [comment]
From what I understand, some changes were made to the API, which causes new avatars to respond with 500 for a day until it's added to a cache database.

> when a new avatar is added it will be on the latest page directly and all following requests for that avatar will fail as error for the next 24h until it'd actually added to the server cache

This does mean that if you see the pending message it doesn't necessarily mean that you were the first to submit it, but it has been within a day of it being submitted.

It doesn't seem like this behavior is going to change soon, so I changed 500 to be a successful submission, change the log level type, and add a pending message to somewhat explain the weird behavior.

**Comment by Chayotea** [comment]
> when a new avatar is added it will be on the latest page directly and all following requests for that avatar will fail as error for the next 24h until it'd actually added to the server cache

You see I wish is that the actual behavior that was happening. In reality is 500 is a 500 forever and never does the avatar appear the latest/database. I have seen hundreds of '500 errors' in the log, which open perfectly in VRCX but are never found in VRCDB. 

Unless I misunderstood somehow this is how I see it. If that quote is what was said to you then it explains why the 500's were dismissed as normal behavior.

**Comment by Chayotea** [comment]
Well last week you updated with a pending message for error 500. Seems the API no longer sends 500 error. What it means for VRC-LOG I guess the pending message is now simply deprecated. From my testing API still seems to be misbehaving it just simply no longer sends a 500.

**Comment by ShayBox** [comment]
I tested myself with a newly uploaded avatar, seems 200 is existing, 404 is new, and I don't get 500 anymore. There's going to be a few more providers coming soon

- [AVTRDB] - [Discord](https://discord.gg/ZxB6w2hGfU) / [Web](https://avtrdb.com) / [VRCX](vrcx://addavatardb/https://api.avtrdb.com/v1/avatar/search/vrcx)
- [DOUGHNUT] - [Discord](https://discord.gg/4HxcPk9r) / [Web](https://avtr1.nekosunevr.co.uk/search.php) / [VRCX](vrcx://addavatardb/https://avtr1.nekosunevr.co.uk/vrcx_search.php)
- [JEFF] - [Discord](https://discord.gg/4HxcPk9r) / [Web](https://avtr.frensmp.cc/search.php) / [VRCX](vrcx://addavatardb/https://avtr.frensmp.cc/vrcx_search.php)
- [NEKO] - [Discord](https://discord.gg/4HxcPk9r) / [Web](https://avtr.nekosunevr.co.uk/search.php) / [VRCX](vrcx://addavatardb/https://avtr.nekosunevr.co.uk/vrcx_search.php)

[AVTRDB]: https://avtrdb.com
[DOUGHNUT]: https://avtr.nekosunevr.co.uk
[JEFF]: https://avtr.frensmp.cc
[NEKO]: https://avtr.nekosunevr.co.uk

**Comment by Chayotea** [comment]
When I wrote the last message the discord bot still responded with  "The application did not respond". Testing in the last few weeks I can officially say it has been fixed. All the failed 500 error avatars I have saved for testing are submitting instantly. My own test avatar finally got submitted after almost 4 months.

Very interesting to see new providers pop up AVTRDB looks really interesting. Even though this was a not a VRC-LOG bug but a API issue I can say this is definitely time to close this.
